src/libsystemd/sd-device/sd-device.c

```diff
@@ -1968,20 +1967,17 @@ _public_ int sd_device_set_sysattr_value(sd_device *device, const char *sysattr,
 
         path = prefix_roota(syspath, sysattr);
 
-        len = strlen(_value);
+        value = strdup(_value);
+        if (!value)
+                return -ENOMEM;
 
         /* drop trailing newlines */
-        while (len > 0 && _value[len - 1] == '\n')
-                len --;
+        truncate_nl(value);
```

Comment:
truncate_nl() truncates at first NL, while the above dropped trailing newline.

use instead:

```c
delete_trailing_chars(value, "\n");
```